app/hotels/src/allHotels/AllHotels.js

```diff
@@ -2,49 +2,99 @@
 
 import * as React from 'react';
 import { graphql } from 'react-relay';
+import { View, StyleSheet } from 'react-native';
 import { PublicApiRenderer } from '@kiwicom/react-native-app-relay';
+import moment from 'moment';
 
+import SearchForm from './SearchForm';
 import AllHotelsSearchList from './AllHotelsSearchList';
 
 import type { AllHotelsSearchQueryResponse } from './__generated__/AllHotelsSearchQuery.graphql';
+import type { Search } from './types';
+
+const styles = StyleSheet.create({
+  wrapper: {
+    flex: 1,
+  },
+});
 
 type Props = {|
   onGoToSingleHotel: () => void,
 |};
 
-export default class AllHotelsSearch extends React.Component<Props> {
+type State = {|
+  search: Search,
+|};
+
+export default class AllHotelsSearch extends React.Component<Props, State> {
+  state = {
+    search: {
+      latitude: 50.08,
+      longitude: 14.44,
+      checkin: null,
+      checkout: null,
+      roomsConfiguration: {
+        adultsCount: 1,
+      },
+    },
+  };
+
+  _handleSearchChange = (search: Search) => {
+    this.setState({
+      search: {
+        latitude: 50.08,
+        longitude: 14.44,
+        checkin: search.checkin,
+        checkout: search.checkout,
+        roomsConfiguration: search.roomsConfiguration,
+      },
+    });
+  };
+
+  isReadyToSearch = () => {
+    const { search } = this.state;
+    return search.checkin && search.checkout;
+  };
+
   renderInnerComponent = (propsFromRenderer: AllHotelsSearchQueryResponse) => (
     <AllHotelsSearchList
       data={propsFromRenderer.allHotels}
       onGoToSingleHotel={this.props.onGoToSingleHotel}
     />
   );
 
-  render = () => (
-    <PublicApiRenderer
-      query={graphql`
-        query AllHotelsSearchQuery($search: HotelsSearchInput!) {
-          allHotels(search: $search) {
-            ...AllHotelsSearchList
-          }
-        }
-      `}
-      variables={{
-        search: {
-          // TODO: values from form
-          latitude: 50.08,
-          longitude: 14.44,
-          checkin: '2018-01-01',
-          checkout: '2018-01-15',
-          roomsConfiguration: {
-            adultsCount: 1,
-          },
-        },
-      }}
-      render={this.renderInnerComponent}
-      cacheConfig={{
-        force: true,
-      }}
-    />
-  );
+  render = () => {
+    const { search } = this.state;
+    return (
+      <View style={styles.wrapper}>
+        <SearchForm onChange={this._handleSearchChange} />
+        {this.isReadyToSearch() && (
```

Comment:
I think some kind of default search would be better than the empty page.

Reply:
Added default search.